### DISCUSSION TITLE
Export failed v2.0.3

### DISCUSSION THREAD

**Discussion by wacomoto** [Zettlr/Zettlr#2809] [Q&A] [answered]
Hi.

I'm getting an ```Export failed``` when exporting to everything I've tried so far: rtf, odt, pdf, html, pdf(latex). 

Things I've tried:

  1. Turned internal pdf off/on.
  2. Uninstalled Zettlr and deleted all user config files. Reinstalled.
  3. Uninstalled Pandoc.
  4. Reinstalled Pandoc. 

OS is Win10. Error window:

```
An error occurred on export: Error running filter C:\Users\NAME\AppData\Roaming\Zettlr\lua-filter\links.lua:


Error running filter C:\Users\NAME\AppData\Roaming\Zettlr\lua-filter\links.lua:
PandocLuaError "table expected, got userdata\n\twhile retrieving list"
stack traceback:
[C]: in field 'walk_block'
C:\Users\NAME\AppData\Roaming\Zettlr\lua-filter\links.lua:47: in function <C:\Users\NAME\AppData\Roaming\Zettlr\lua-filter\links.lua:41>
```

**Comment by nathanlesage** [comment, marked as answer]
Oh, I made an error in the links script. Will tend to this; until then: Do not activate "Remove internal links completely" in the export settings, that will prevent the error.

**Reply by nathanlesage** [reply]
Fixed, will ship with next update

**Reply by wacomoto** [reply]
That did the trick. Thanks Hendrik!

**Comment by CornelisVI** [comment]
I have exactly the same error:

"An error occurred on export: Error running filter /home/drcor/.config/Zettlr/lua-filter/links.lua:

Error running filter /home/drcor/.config/Zettlr/lua-filter/links.lua:
module 'lpeg' not found:
lpeg is not one of pandoc's default packages
no field package.preload['lpeg']
no file '/usr/share/lua/5.4/lpeg.lua'
no file '/usr/share/lua/5.4/lpeg/init.lua'
no file '/usr/lib/lua/5.4/lpeg.lua'
no file '/usr/lib/lua/5.4/lpeg/init.lua'
no file './lpeg.lua'
no file './lpeg/init.lua'
no file '/usr/lib/lua/5.4/lpeg.so'
no file '/usr/lib/lua/5.4/loadall.so'
no file './lpeg.so'
lpeg was configured to be excluded from the binary."

Tried to uninstall everything and install it again, deleted the .config folder for zettlr. Uninstall > reinstall. All to no avail. 
My Zettlr version is 2.3.0 on Manjaro Linux KDE. Aur activated.

**Reply by CornelisVI** [reply]
I tried also to refresh the BetterBixTex files *.bib or *.json. They offered different errors as a result. The *.bib file produced the same as I mentioned, the *.json version spawned the following:
"An error occurred on export: Error reading bibliography file /home/drcor/Nextcloud/DOCTORAAT/PUBLICATIES/2021_HERWERKING/DEEL_I/HOOFDSTUK2/MASCULINE_DANCE2.json:

Error reading bibliography file /home/drcor/Nextcloud/DOCTORAAT/PUBLICATIES/2021_HERWERKING/DEEL_I/HOOFDSTUK2/MASCULINE_DANCE2.json:
Error in $: parsing [] failed, expected Array, but encountered Object"

I think that basically the lua 5.4 / lpeg dependency is what goes wrong here. 
There are other mentions about this failing dependency elsewhere:

https://bugs.archlinux.org/task/75296
Any suggestions for what I could do?
Kind regards,
Cor
